podcasts/New Search/Helpers/SearchAnalyticsHelper.swift

```diff
@@ -0,0 +1,85 @@
+import SwiftUI
+import PocketCastsServer
+
+class SearchAnalyticsHelper: ObservableObject {
+    let source: AnalyticsSource
+
+    init(source: AnalyticsSource) {
+        self.source = source
+    }
+
+    // MARK: - Search
+
+    func trackShown() {
+        Analytics.track(.searchShown, properties: ["source": source])
+    }
+
+    func trackSearchPerformed() {
+        Analytics.track(.searchPerformed, properties: ["source": source])
+    }
+
+    func trackFailed() {
+        Analytics.track(.searchFailed, properties: ["source": source])
+    }
+
+    func trackResultTapped(_ searchResult: Any) {
+        var type = "unknown"
+        var uuid = ""
+        if let folderOrPodcast = searchResult as? PodcastFolderSearchResult {
+            type = folderOrPodcast.type
+            uuid = folderOrPodcast.uuid
+        } else if let episode = searchResult as? EpisodeSearchResult {
+            type = "episode"
+            uuid = episode.uuid
+        }
+        Analytics.track(.searchResultTapped, properties: ["source": source, "uuid": uuid, "result_type": type])
+    }
+
+    // MARK: - Search History
+
+    func trackHistoryCleared() {
+        Analytics.track(.searchHistoryCleared, properties: ["source": source])
+    }
+
+    func historyItemTapped(_ entry: SearchHistoryEntry) {
+        let uuid = entry.podcast?.uuid ?? entry.episode?.uuid ?? ""
+        Analytics.track(.searchHistoryItemTapped, properties: ["source": source, "uuid": uuid, "type": entry.type])
```

Comment:
We could use the `AnalyticsSearchResultItem` protocol here too:

```
extension SearchHistoryEntry: AnalyticsSearchResultItem {
    var uuid: String {
        podcast?.uuid ?? episode?.uuid ?? ""
    }

    var analyticsDescription: String {
       ...
    }
}
```


```suggestion
        Analytics.track(.searchHistoryItemTapped, properties: ["source": source, "uuid": entry.uuid, "type": entry])
```
```